packages/xdl/src/Web.js

```diff
@@ -15,6 +17,38 @@ function invokePossibleFunction(objectOrMethod, ...args) {
   }
 }
 
+export async function ensureDevPackagesInstalledAsync(projectRoot, ...newDevDependencies) {
+  const { exp } = await readConfigJsonAsync(projectRoot);
+
+  // Filter out packages that are already installed.
+  let modulesToInstall = [];
+  for (const module of newDevDependencies) {
+    if (!(await ConfigUtils.resolveModule(module, projectRoot, exp))) {
+      modulesToInstall.push(module);
+    }
+  }
+
+  const useYarn = await fs.exists(path.resolve('yarn.lock'));
+  if (useYarn) {
+    console.log('Installing packages with yarn...');
+    const args = modulesToInstall.length > 0 ? ['add', '--dev', ...modulesToInstall] : [];
+    spawn.sync('yarnpkg', args, { stdio: 'inherit' });
```

Comment:
Please check out [`PackageManager.js`](https://github.com/expo/expo-cli/blob/f9b90874f36b8c1d185d32c5ab2d01ee2fe0d2f5/packages/expo-cli/src/PackageManager.js). I made it for `expo install` and I think we could centralize our npm/Yarn usage to it instead of writing the code for spawning these commands every time. It also includes some goodies like not printing those obnoxious peerDependency warnings.

How to use:
```js
const packageManager = PackageManager.createForProject(projectRoot);

await packageManager.addAsync(...modulesToInstall);
```
For `devDependencies`, you could add `PackageManager.addDevAsync` that uses the `--dev`/`--save-dev` setting.